### PR TITLE
CI: add macOS arm runners and update macOS CI in general 

### PR DESCRIPTION
**Pull request by julianoes**
Let's see what happens.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -515,7 +533,11 @@ jobs:
...
         with:
           name: mavsdk_server_macos.framework
-          path: ./build/macos/src/mavsdk_server/src/mavsdk_server.framework
+          path: ./build/macos/src/mavsdk_server/src/mavsdk_server_x64.framework
+      - uses: actions/download-artifact@v4
+        with:
+          name: mavsdk_server_macos.framework
+          path: ./build/macos/src/mavsdk_server/src/mavsdk_server_arm64.framework
       - name: Package
         run: bash ./src/mavsdk_server/tools/package_mavsdk_server_framework.bash
```
Comment (JonasVautherin) [inline .github/workflows/main.yml:542]: Note that this script will have to be updated if you want to include the new builds in the final framework :+1:

Comment (julianoes) [reply .github/workflows/main.yml:542]: Yes, do you know how? I couldn't really find the docs on how the structure should be.

Comment (JonasVautherin) [reply .github/workflows/main.yml:173]: I think what you did (after the comment) makes sense, but I don't really understand the other changes. Was there a need to update the dependencies and toolchain for this?

File: tools/ios.toolchain.cmake [hunk 2]
```diff
@@ -151,10 +151,10 @@
...
 
 # CMake invokes the toolchain file twice during the first build, but only once during subsequent rebuilds.
-#if(DEFINED ENV{_IOS_TOOLCHAIN_HAS_RUN})
-#  return()
-#endif()
-#set(ENV{_IOS_TOOLCHAIN_HAS_RUN} true)
+if(DEFINED ENV{_IOS_TOOLCHAIN_HAS_RUN})
+  return()
+endif()
+set(ENV{_IOS_TOOLCHAIN_HAS_RUN} true)
```
Comment (JonasVautherin) [inline tools/ios.toolchain.cmake:157]: Pretty sure you don't want this. The maintainer of the toolchain file doesn't remember why he added this and refuses to revert it, and it has always been a problem for us. See https://github.com/leetal/ios-cmake/pull/198 and https://github.com/leetal/ios-cmake/issues/146.

Comment (julianoes) [reply tools/ios.toolchain.cmake:157]: Oh, right, thanks!

Comment (julianoes) [reply tools/ios.toolchain.cmake:157]: So it didn't actually update at all, ok I'm stupid.

File: .github/workflows/main.yml (hunk 1 above)
Comment (julianoes) [reply .github/workflows/main.yml:173]: I'm working on it...

**Comment by julianoes** [comment]
@JonasVautherin I'm lost, any idea?
https://github.com/mavlink/MAVSDK/actions/runs/8903079132/job/24451638181?pr=2277#step:9:1604

**Comment by julianoes** [comment]
I think I need to buy a Mac, just so I can throw it out the window.

**Comment by julianoes** [comment]
I will have to rent a Mac in the cloud again and try it...

**Comment by julianoes** [comment]
Hurray! Thanks for the help @JonasVautherin!

**Pull request merged**